MAINTAINERS.md

```diff
@@ -0,0 +1,197 @@
+
+
+## Current Maintainers
+
+The current maintainers are:
+
+| name | focus areas |
+------------------
+|   @achow101  | wallet, PSBTs |
+|   @fanquake | general, tooling, build system |
+|   @hebasto | UI/UX |
+|   @laanwj | general |
+|   @marcofalke | general |
+|   @sipa | general, peer-to-peer, consensus, cryptography |
+
+See [REVIEWERS](/REVIEWERS) for more information on specific focus areas of
+maintainers and regular contributors.
+
+
+## Regular Contributors, Scoped Maintainers, General Maintainers, and Lead Maintainers
+
+The title of Maintainer in Bitcoin is more of a "janitorial responsibility"
+than a position of privilege. However, the role does include permissions on the
+repository, and connotes a general sense of stewardship over the project.
+
+
+Largely, the role of maintainers is to decide to merge Pull Requests. This
+process is described in [CONTRIBUTING.md](/CONTRIBUTING.md), but as an excerpt:
+
+> Whether a pull request is merged into Bitcoin Core rests with the project merge
+> maintainers.
+>
+> Maintainers will take into consideration if a patch is in line with the general
+> principles of the project; meets the minimum standards for inclusion; and will
+> judge the general consensus of contributors.
+
+Thus maintainers role is not directly elevated above that of regular
+contributors, as maintainers serve mostly as arbiters of if changes are
+acceptable to the contributors to the project in aggregate.
+
+The role of the Maintainer is currently evolving with respect to "Scoped
+Maintainers". The concept of a scoped maintainer is that it is helpful to add
+maintainers who are subject matter experts in a particular area of the code,
+but perhaps not in other areas. As such, maintainers with an explicit scope
+typically will avoid merging PRs in areas outside their scope, but may do so on
+occasion. As an individual maintainer's experience levels in the project
+increase, they may change their focus from scoped to general, but this is not
+currently an explicit designation (all maintainers should be considered to be
+general at this time). As a rough measure, scoped maintainers should avoid
+"stepping on the toes" of work that is being guided by another maintainer or
+contributor.
+
+Historically, Bitcoin has also had a "Lead Maintainer", which was most recently
+@laanwj. However, [in a blog
+post](https://laanwj.github.io/2021/01/21/decentralize.html), @laanwj stated
+that they wanted to take more of a background role as a maintainer and have not
+"passed the baton" to a successor as a Lead Maintainer, and instead focused on
+decentralizing the maintainership ecosystem. As such, while the title of Lead
+Maintainer still is held by @laanwj, it should be considered a bug-not-feature
+if it confers any additional responsibility or privilege over other General
+Maintainers, and may be explicitly deprecated in the future.
+
+## Self Merges
+
+It is discouraged for maintainers to merge their own work. However, given that
```

Comment:
Why is it discouraged? The same criteria should apply...

Reply:
I think there are a number of reasons for it to be discouraged, but I agree the same criteria ultimately applies: it has to be good code for Bitcoin ;)

I think there are a couple practical concerns, one being that on the bubble self merges might be less well reviewed (e.g. is the self merge happening because no one else is qualified to review it?) and also a centralizing issue whereby maintainers might be much more "productive" than regular contributors because they favour their own work over the work of others.

Reply:
**Why self merges should be discouraged?**

- Reviewers can find better approach, bugs etc. in code because a developer who writes code may not be able to review own code without bias. Similarly, other maintainer merging the pull request would take the responsibility to make changes, will be less biased and go through everything before merging.

- If other maintainer who merges code did not already review the pull request, it will be helpful in getting another review and find new issues, bugs etc. that may go unnoticed. 

Example: If laanwj decides to merge https://github.com/bitcoin/bitcoin/pull/24362 instead of achow101 there will be another review, no bias and maybe new issues, bugs etc.

Only positive of allowing self merges is merge/week or merge/month is not affected even if less active maintainers.

Reply:
> every criticism of a self-merge that i have seen so far really boils down to "merging a PR that has not received sufficient review or consensus

No, it is different from merging a less reviewed pull request. I have shared 2 reasons above your comment:

1. **Bias**: Alice and Bob are maintainers. Alice opens a pull request which is reviewed by others. Maintainers are humans and assessment of the reviews for this pull request can be different by Alice and Bob. Alice will have some bias to get it merged with 3 ACKs and Bob will look at all the comments, reviews, nits, possibility of bugs etc. before merging.

2. **1 Extra review before merge:** In some cases it adds another review before merge by another maintainer. More reviews should never be a problem.

Reply:
Regardless of how "weakly" they are discouraged, they will decrease transparency, as the slightest discouragement will encourage maintainers to cooperate on trading merges privately.

Reply:
Transparency isn't affected because everyone can see who merged the pull request. Private communications cannot be stopped and there is no way to verify if they already affect pull requests. Things that are written publicly and merging the pull requests is important. If someone wants to take the responsibility of merging a pull request it doesn't matter if there was a private trade involved. If everything is checked before merging without bias it should be okay else its risking reputation.

Reply:
@michaelfolkson

> I'd dial these back if I were you.

Do not expect every contributor in this repository or on GitHub to follow your rules.

> You are not an authority on this pull request.

I am a reviewer in this pull request writing my thoughts and responding to comments.

> Questions and challenges are fine to improve your own understanding but lecturing maintainers on what is and isn't true as a new contributor verges on the bizarre. 

There are no lectures or truths being shared. Opinions of different people about self merges.

> Maintainers know what has worked well and what hasn't in the past and have a much better understanding than you on the trade-offs for suggested changes.

I do not need your approval in every pull request to write something and this isn't the first time you are trying to bully new contributors. Hopefully you would learn something and comment on things being discussed.